lib/types/jtd-schema.ts

```diff
@@ -149,3 +147,67 @@ export type JTDSchemaType<T, D extends Record<string, unknown> = Record<string,
   // TODO these should only be allowed at the top level
   definitions?: {[K in keyof D]: JTDSchemaType<D[K], D>}
 }
+
+type JTDDataDef<S, D extends Record<string, unknown>> =
+  | (// ref
+    S extends {ref: string}
+      ? JTDDataDef<D[S["ref"]], D>
+      : // type
+      S extends {type: NumberType}
+      ? number
+      : S extends {type: "string"}
+      ? string
+      : S extends {type: "timestamp"}
+      ? string | Date
+      : // enum
+      S extends {enum: readonly (infer E)[]}
+      ? string extends E
+        ? never
+        : [E] extends [string]
+        ? E
+        : never
+      : // elements
+      S extends {elements: infer E}
+      ? JTDDataDef<E, D>[]
+      : // properties
+      S extends {
+          properties: Record<string, unknown>
+          optionalProperties?: Record<string, unknown>
+          additionalProperties?: boolean
```

Comment:
Could it treat true separately here by allowing any additional properties with within the data interface?

Reply:
that's a good idea! This should be resolved in the most recent commit.